src/listpack.c

```diff
@@ -1112,6 +1112,44 @@ unsigned char *lpBatchDelete(unsigned char *lp, unsigned char **ps, unsigned lon
     return lpShrinkToFit(lp);
 }
 
+/* Trim a listpack, remove 'ltrim' elements from left and 'rtrim' elements from right. */
+unsigned char *lpTrim(unsigned char *lp, unsigned long ltrim, unsigned long rtrim) {
+    if (ltrim == 0 && rtrim == 0) return lp;
+    size_t bytes = lpBytes(lp);
+    unsigned long llen = lpLength(lp);
+    if (ltrim + rtrim >= llen) {
+        /* Remove all elements */
+        lp[LP_HDR_SIZE] = LP_EOF;
+        lpSetTotalBytes(lp, LP_HDR_SIZE + 1);
+        lpSetNumElements(lp, 0);
+        return lpShrinkToFit(lp);
+    }
+    unsigned long rangelen = llen - ltrim - rtrim;
+    unsigned char *first, *tail; /* first points to the first byte we want to save, 
+                                    and tail points to the byte after the last byte we want to save */
+    if ((first = lpSeek(lp, ltrim)) == NULL) return lp;
```

Comment:
```suggestion
    first = lpSeek(lp, ltrim);
    assert(first != NULL);
```
we should assert instead of hiding this.

Reply:
Commit the suggestion, but I see that in [lpDeleteRange](https://github.com/redis/redis/blob/becd50d0da70cb0f55e6e40317c0d2ff5765a304/src/listpack.c#L1044), we just return lp when lpSeek returns NULL.  Should we change to use assertion in lpDeleteRange?

Reply:
we can't determine if `index` is out of range in `lpDeleteRange`, but we can make sure that `ltrim` isn't out of range.
so `assert` is better than return NULL.

Reply:
You are right, it is better to expose the memory corruption by using assert since we are sure that it isn't out of range